sklearn/decomposition/tests/test_pca.py

```diff
@@ -161,7 +161,9 @@ def test_pca_sparse_fit_transform(global_random_seed, sparse_container):
 
 @pytest.mark.parametrize("svd_solver", ["randomized", "full", "auto"])
 @pytest.mark.parametrize("sparse_container", CSR_CONTAINERS + CSC_CONTAINERS)
-def test_sparse_pca_solver_error(global_random_seed, svd_solver, sparse_container):
+def test_sparse_pca_solver_automatically_select_arpack(
```

Comment:
We keep this test but we need to make sure that it does not raise with `"auto"`.

We also need a new test where `"auto"` should not raise and give the same result as forcing `"arpack"`.